corehq/apps/settings/forms.py

```diff
@@ -181,7 +181,18 @@ def __init__(self, *, allow_phone_2fa, **kwargs):
 
 
 class HQTOTPDeviceForm(TOTPDeviceForm):
-    token = forms.IntegerField(required=False, label=_("Token"), min_value=1, max_value=int('9' * totp_digits()))
+    token = forms.IntegerField(
+        required=False,
+        label=_("Token"),
+        min_value=1,
+        max_value=int('9' * totp_digits()),
+        widget=forms.TextInput(
+            attrs={
+                'pattern': r'^[0-9]{6}$',
```

Comment:
What do you think of this change?

```suggestion
                'pattern': fr'^[0-9]{{{totp_digits()}}}$',
```

Or if that's too many curly braces
```suggestion
                'pattern': r'^[0-9]{%s}$' % totp_digits(),
```

Reply:
I like it. Didn't actually realize you could have formatted regular expression string like that, but the second choice seems nicer to read anyway.